sopherapps/lilly
# v0.2.0 Add CRUD Actions

## Done: [0.2.0] - 2022-05-06

### Added

- `CreateOneAction` for creating a single item in the respective repository
- `CreateManyAction` for creating multiple item in the respective repository
- `UpdateOneAction` for updating a single item from the respective repository
- `UpdateManyAction` for updating multiple item in the respective repository
- `DeleteOneAction` for deleting a single item from the respective repository
- `DeleteManyAction` for deleting multiple items from the respective repository
- `ReadOneAction` for reading a single item from the respective repository
- `ReadManyAction` for reading many items from the respective repository
- `create_name` route to the stub
- `create_many_name` route to the stub
- `update_name` route to the stub
- `update_many_names` route to the stub
- `update_name_partially` route to the stub
- `read_many_names` route to the stub
- `read_name` route to the stub
- `delete_many_names` route to the stub
- `delete_name` route to the stub

### Changed

- The type of the `record`/`new_record` parameter in all repository major methods that need it has changed from `Any`
  to `BaseModel`.
- The type of the `record`/`new_record` parameter in all action major methods that need it has changed from `Any`
  to `BaseModel`.

### Fixed

- Filtering using text SQL in sqlalchemy was fixed by initially coercing them using `text(string_sql)` and then ensuring
  `update` and `delete` statements use `execution_options.synchronize_session="fetch"`

## How to test

- Run the unit tests for all

```shell
python -m unittest
```
- To see it in action, follow the instructions on the README to create a new app and run the app and open the docs endpoint in your browser according to the instructions. 
- Open [http://127.0.0.1:8000/docs](http://127.0.0.1:8000/docs) in your browser and try out any of the endpoints on the OpenAPI docs shown
